x509-identity-mgmt/js/src/app.js

```diff
@@ -0,0 +1,128 @@
+const express = require('express');
+
+const addRequestId = require('express-request-id');
+
+const compression = require('compression');
+
+const createError = require('http-errors');
+
+const morgan = require('morgan');
+
+const path = require('path');
+
+const { Logger } = require('@dojot/microservice-sdk');
+
+const serveIndex = require('serve-index');
+
+const jwtDecode = require('jwt-decode');
+
+const cfg = require('./config');
+
+const { pagingMiddleware } = require('./core/paginate');
+
+/* This library is about what happens when you hit an async error. */
+require('express-async-errors');
+
+Logger.setTransport('console');
+
+const logger = new Logger();
+
+const app = express();
+
+/* Generate UUID for request and add it to X-Request-Id header.
+ * In case request contains X-Request-Id header, uses its value instead. */
+app.use(addRequestId());
+
+/* When running an Express app behind a proxy */
+if (cfg.trustProxy) {
+  app.set('trust proxy', true);
+}
+
+/* compress all responses */
+app.use(compression());
+
+/* setup the logger */
+logger.stream = {
+  write: (message) => {
+    logger.info(message);
+  },
+};
+morgan.token('id', (req) => req.id);
+
+app.use(morgan(cfg.logFormat, { stream: logger.stream }));
+
+/* Use a middleware that parses JSON and only looks at requests
+ * where the Content-Type header matches the type "application/json"
+ * It also controls the maximum request body size by "limit" option,
+ * the value is passed for parsing to the Bytes Library:
+ * https://www.npmjs.com/package/bytes */
+app.use(express.json(cfg.bodyParser));
+
+/* Paginate middleware */
+app.use('/', pagingMiddleware(cfg.paginate.limit, cfg.paginate.maxLimit));
+
+/* Decodes the JWT token that must be sent with the request.
+ * The token validation is not performed, as it is expected
+ * to be validated by the API Gateway. */
+app.use((req, res, next) => {
+  const err = new createError.Unauthorized();
+  if (req.path.includes('throw-away')) {
+    return next();
```

Comment:
It is possible to reach this instruction externally to skip JWT decode step ?, for example putting "throw-away" in the URL ?

Reply:
The idea of this middleware is to apply to all routes, even those that may arise in the future. 
In this case, the "throw-away" route is the only exception, as it is used by the services behind the API Gateway and there is no need for the tenant's data.. 
I also did not like the way it is, since we have not yet planned how it will be to obtain the certificate by microservices. 
I think that will change in the future when we plan this feature.
For now, this is really a "workaround".

Reply:
Ok, I got it.